Gems/Atom/RPI/Code/Include/Atom/RPI.Edit/Material/MaterialSourceData.h

```diff
@@ -59,24 +59,25 @@ namespace AZ
 
             uint32_t m_materialTypeVersion = 0; //!< The version of the material type that was used to configure this material
 
-            struct Property
-            {
-                AZ_TYPE_INFO(AZ::RPI::MaterialSourceData::Property, "{8D613464-3750-4122-AFFE-9238010D5AFC}");
-
-                MaterialPropertyValue m_value;
-            };
-
-            using PropertyMap = AZStd::map<AZStd::string, Property>;
-            using PropertyGroupMap = AZStd::map<AZStd::string, PropertyMap>;
-
-            PropertyGroupMap m_properties;
-
             enum class ApplyVersionUpdatesResult
             {
                 Failed,
                 NoUpdates,
                 UpdatesApplied
             };
+            
+            //! If the data was loaded from an old format file (i.e. where "properties" was a tree with property values nested under groups),
+            //! this converts to the new format where properties are stored in a flat list.
+            void ConvertToNewDataFormat();
+
+            // Note that even though we use an unordered map, the JSON serialization system is nice enough to sort the data when saving to JSON.
+            using PropertyValueMap = AZStd::unordered_map<Name, MaterialPropertyValue>;
+
+            void SetPropertyValue(const Name& propertyId, const MaterialPropertyValue& value);
+            const MaterialPropertyValue& GetPropertyValue(const Name& propertyId) const;
+            PropertyValueMap GetPropertyValues() const;
```

Comment:
Somewhat of a nit, but for getters that do a possibly more expensive copy, I tend to prefer something like `ClonePropertyValues` (also as a reminder that you aren't getting a reference since this returned value could bind to a `const PropertyValueMap&` and compile also)

Reply:
Oh, thanks for catching this! I was using PropertyValueMap temporarily at an early state where I had to construct the map, and meant to go back and change to a reference but forgot.